docs/docusaurus/docs/cloud/connect/connect_snowflake.md

```diff
@@ -25,56 +23,74 @@ New to GX Cloud and not sure that it's the right solution for your organization?
 
-1. In Snowflake Snowsight, click **Worksheets** > **Add** > **SQL Worksheet**.
+1. In GX Cloud, click **Data Assets** > **New Data Asset**.
+
+2. Copy the code in the code pane.
+
+3. Prepare your Snowflake environment:
+
+   - In Snowflake Snowsight, click **Worksheets** > **Add** > **SQL Worksheet**.
+
+   - Paste the code you copied in step 2 into the SQL worksheet.
+
+      Replace `secure_password` with your value and `database_name` and `schema_name` with the names of the database and schema you want to access in GX Cloud.
+
+      `grant select on future tables in schema "database_name.schema_name" to role gx_role;` is optional and gives the user with the `gx_role` role access to all future tables in the defined schema.
+
+      The settings in the code example optimize cost and performance. Adjust them to suit your business requirements.
+
+   - Select **Run All** to define your user password, create a new GX role (`gx_role`), assign the password and role to a new user (`gx_user`), create a new warehouse (`gx_wh`) for the GX Agent, and allow the user with the `gx_role` role to access data on the Snowflake database and schema.
+
+       ![Snowflake Run All](/img/run_all.png)
+
+4. In GX Cloud, click **I have created a GX Cloud user with valid permissions** and then click **Continue**.
+
+5. Enter a meaningful name for the Data Asset in the **Data Source name** field.
+
+6. Optional. To use a connection string to connect to a Data Source, click the **Use connection string** selector, enter a connection string, and then move to step 6. The connection string format is: `snowflake://<user_login_name>:<password>@<accountname>`.
+
+7. Complete the following fields:
+
+    - **Username**: Enter the username you use to access Snowflake.
+
+    - **Account identifier**: Enter your Snowflake organization and account name separated by a hyphen (`oraganizationname-accountname`) or your account name and a legacy account locator separated by a period (`accountname.region`). The legacy account locator value must include the geographical region. For example, `us-east-1`. 
+    
+        To locate your Snowflake organization name, account name, or legacy account locator values see [Finding the Organization and Account Name for an Account](https://docs.snowflake.com/en/user-guide/admin-account-identifier#finding-the-organization-and-account-name-for-an-account) or [Using an Account Locator as an Identifier](https://docs.snowflake.com/en/user-guide/admin-account-identifier#using-an-account-locator-as-an-identifier).
+
+    - **Password**: Enter a Snowflake password. To improve data security, GX recommends using a Snowflake service account to connect to GX Cloud.
 
-2. Copy and paste the following code into the SQL worksheet:
+    - **Database**: Enter the name of the Snowflake database where the data you want to validate is stored. In Snowsight, click **Data** > **Databases**. In the Snowflake Classic Console, click **Databases**.
```

Comment:
Here we're telling people to click things, but not telling them why or what to do with what they get back.  Perhaps:
```suggestion
    - **Database**: Enter the name of the Snowflake database where the data you want to validate is stored. To find the names of available databases in Snowsight, click **Data** > **Databases**. In the Snowflake Classic Console, find the available database names by clicking **Databases**.
```

Reply:
@Rachel-Reverie, if you're in the application, it is clear what you're doing. Also, the user is instructed to "enter the name of the Snowflake database", so it's safe to assume that they know what the instructions are for.